### ISSUE TITLE
object-curly-spacing: miss-handles type annotations

### ISSUE THREAD

**Issue by zaygraveyard** [eslint/eslint#4752]
### test.js
```js
/*eslint object-curly-spacing: [2, "never"] */

function fnObject({key1, key2}: Object): void {
  // nothing important
}
```

### What versions I'm using
eslint@1.10.3
babel-eslint@4.1.6

### What I did
```sh
$eslint --parser babel-eslint test.js
```

### What I expected
No errors

### What actually happened
```
/tmp/test.js
  3:33  error  There should be no space before 'Config'  object-curly-spacing

✖ 1 problem (1 error, 0 warnings)
```

### My Observations

The `ObjectPattern` node of the AST tree contains the type annotation.

```js
function fnObject({key1, key2}: Object): void {
//                ^------------------^
//                   ObjectPattern
```

The `object-curly-spacing` rule checks the last and before last tokens of the AST node (ie. last is `Object` an before last is `:`).

Since the between the before last token and last token their is a space and the expected is no space, an error is reported.

**Comment by zaygraveyard** [comment]
Working on it

**Comment by ilyavolodin** [comment]
ESLint doesn't support experimental features and flow. You might want to check out [eslint-plugin-babel](https://github.com/babel/eslint-plugin-babel).

**Comment by zaygraveyard** [comment]
I didn't know about this plugin, I'll check it out.
Thanks @ilyavolodin.

Should this issue be closed?

**Comment by ilyavolodin** [comment]
Yes. Sorry, we have no plans to support flow as part of the core.
